ts/customEvents.ts

```diff
@@ -0,0 +1,253 @@
+		chart.series.forEach(series => {
+			bindElementEvents(series.dataLabelsGroup, series.options.dataLabels?.events, chart._customEventsBound);
+		});
+	}
+
+	/**
+	 * Unified lifecycle handler for chart load and redraw events
+	 * 
+	 * This function handles both initial load and redraw scenarios by cleaning up
+	 * existing events and rebinding to all elements (including new ones).
+	 * 
+	 * @param {Chart} this - The chart instance
+	 * @returns {void}
+	 */
+	const lifecycleHandler = function (this: Chart) {
+		if (this._customEventsBound) {
+			cleanupEvents(this._customEventsBound);
+		}
+		bindChartEvents(this);
+	};
+
+	H.addEvent(H.Chart, "load", lifecycleHandler);
+	H.addEvent(H.Chart, "redraw", lifecycleHandler);
+
+	/**
+	 * Chart destroy event handler that cleans up all bound events
+	 * 
+	 * This function is called when a chart is destroyed to prevent memory leaks
+	 * by removing all custom event listeners.
+	 * 
+	 * @param {Chart} this - The chart instance
+	 * @returns {void}
+	 */
+	H.addEvent(H.Chart, "destroy", function (this: Chart) {
+		const chart = this;
+
+		// Clean up all bound events
+		if (chart._customEventsBound) {
+			cleanupEvents(chart._customEventsBound);
+			delete chart._customEventsBound;
+		}
+	});
+
+	/**
+	 * Wrap Axis.addPlotBand and Axis.addPlotLine to bind events to new plot bands/lines
+	 * 
+	 * This ensures that dynamically added plot bands and lines get event bindings.
+	 */
+	if (H.Axis && H.Axis.prototype) {
+		// Wrap addPlotBand
+		H.wrap(H.Axis.prototype, 'addPlotBand', function (this: Highcharts.Axis, proceed: Function, options: Highcharts.AxisPlotBandsOptions) {
+			const result: Highcharts.PlotLineOrBand = proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+
+			// Bind events to the new plot band if it has a label
+			if (result && result.label && this.chart) {
+				bindElementEvents(result.label, options.label?.events, this.chart._customEventsBound);
+			}
+
+			return result;
+		});
+
+		// Wrap addPlotLine
+		H.wrap(H.Axis.prototype, 'addPlotLine', function (this: Highcharts.Axis, proceed: Function, options: Highcharts.AxisPlotBandsOptions) {
+			const result: Highcharts.PlotLineOrBand = proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+
+			// Bind events to the new plot line if it has a label
+			if (result && result.label && this.chart) {
+				bindElementEvents(result.label, options.label?.events, this.chart._customEventsBound);
+			}
+
+			return result;
+		});
+
+		// TODO: This needs to be taken care of
+		H.wrap(H.Axis.prototype, 'drawCrosshair', function (
+			this: Highcharts.Axis,
+			proceed: (e?: Highcharts.PointerEventObject, point?: Highcharts.Point) => void,
+			e?: Highcharts.PointerEventObject,
+			point?: Highcharts.Point
+		): void {
+			console.log('calling wrap!');
+
+			proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+
+			if (this.cross && this.crosshair && this.chart && (this.chart as any)._customEventsBound) {
```

Comment:
Why `as any` if the types are extended like `  interface Chart {
    _customEventsBound?: BoundEvent[];
  }` ?